tests/runner/src/env.rs

```diff
@@ -635,3 +739,117 @@ impl Display for ResultDisplayer {
         }
     }
 }
+
+struct PostgresqlFormatter {
+    pub rows: Vec<PgRow>,
+}
+
+impl Display for PostgresqlFormatter {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if self.rows.is_empty() {
+            return f.write_fmt(format_args!("(Empty response)"));
+        }
+
+        // create schema
+        let schema = match &self.rows[0] {
+            PgRow::CommandComplete(affected_rows) => {
+                write!(
+                    f,
+                    "{}",
+                    ResultDisplayer {
+                        result: Ok(Output::new_with_affected_rows(*affected_rows as usize)),
+                    }
+                )?;
+                return Ok(());
+            }
+            PgRow::RowDescription(desc) => Arc::new(Schema::new(
+                desc.iter()
+                    .map(|column| {
+                        ColumnSchema::new(column.name(), ConcreteDataType::string_datatype(), false)
+                    })
+                    .collect(),
+            )),
+            _ => unreachable!(),
+        };
+        if schema.num_columns() == 0 {
+            return Ok(());
+        }
+
+        // convert to string vectors
+        let mut columns: Vec<StringVectorBuilder> = (0..schema.num_columns())
+            .map(|_| StringVectorBuilder::with_capacity(schema.num_columns()))
+            .collect();
+        for row in self.rows.iter().skip(1) {
+            if let PgRow::Row(row) = row {
+                for (i, column) in columns.iter_mut().enumerate().take(schema.num_columns()) {
+                    column.push(row.get(i));
```

Comment:
Is it possible that the output column doesn't have the string type?

Reply:
No. Pg's simple query protocol return [every retried data in text](https://www.postgresql.org/docs/current/protocol-flow.html#PROTOCOL-FLOW-SIMPLE-QUERY:~:text=In%20simple%20Query%20mode%2C%20the%20format%20of%20retrieved%20values%20is%20always%20text%2C%20except%20when%20the%20given%20command%20is%20a%20FETCH%20from%20a%20cursor%20declared%20with%20the%20BINARY%20option.%20In%20that%20case%2C%20the%20retrieved%20values%20are%20in%20binary%20format.%20The%20format%20codes%20given%20in%20the%20RowDescription%20message%20tell%20which%20format%20is%20being%20used.).